Corrections to migration guide (#184)
valkey-io/valkey-doc: topics/migration.md

### Migration compatibility matrix

You can migrate a Redis server to Valkey.
Valkey is compatible with Redis OSS 7.2 and all earlier open source Redis versions.

> Migrating from any open source Redis version to Valkey is just an upgrade.

Reviewer: I would change 'just' -> 'effectively'

Author: ```suggestion
Migrating from any open source Redis version to Valkey is effectively an upgrade.
```

> Valkey is backward compatible with Redis OSS 7.2 and all earlier open source Redis versions.

Reviewer 2: ```suggestion
Valkey is compatible with Redis OSS 7.2 and all earlier open source Redis versions.
```
It's a different engine, it's not backwards compatibility.

> Redis Community Edition (CE), versions 7.4 and later, are not open source and the data files are not compatible with Valkey.

Reviewer: I assume you mean RDB files? I don't think we use the term 'data files' often. If there is a link that explains the difference between 7.4 and Valkey that would be helpf to the reader.

Reviewer 2: It would also include AOF files I presume, so not just RDB.

Author: I don't have time to go into the details how you can migrate. It's possible to use various tools, like rdblib and RediShake, that can convert between different RDB versions, connect as a replica and emit commands to a Valkey node, or something like that. To do this without any service impact is probably possible, but I don't have enough experience with it.

I just want to remove the parts that are incorrect in this document, and to make it not imply that it's impossible to migrate between any other versions than those listed in the table.

Author: Btw, Kyle, I agree with your comments. It's just a matter of prioritization.

> Redis 7.4 and later are not open source and the data files are no longer compatible with Valkey.

Reviewer 2: ```suggestion
Redis Community Edition (CE), versions 7.4 and later, are not open source and the data files are not compatible with Valkey.
```

It may be possible to migrate the data to Valkey from proprietary Redis versions and other Redis-like software but it requires another method and is not covered by this document.

> The following table provides migration options depending on the open source Redis version you run:
> 
> | Redis                 | Valkey |
> |-----------------------|--------|
> | 2.x - 7.2.x           | 7.2.x  |
> | 2.x - 7.2.x           | 8.0    |
> | 7.4 (not open source) | n/a    |

Reviewer 2: ```suggestion
The following table provides migration options depending on the Redis version you run:

| Redis                 | Valkey |
|-----------------------|--------|
| OSS 2.x - 7.2.x       | 7.2.x  |
| OSS 2.x - 7.2.x       | 8.0    |
| CE 7.4                | n/a    |
```

I found it odd that you added open source in the header, but then later called it non-OSS. Maybe this is clearer?
